src/D2L.CodeStyle.Analyzers/Extensions/Microsoft.CodeAnalysis.cs

```diff
@@ -216,5 +216,32 @@ string name
 
 			return false;
 		}
+
+
+		/// <summary>
+		/// Check if the symbol has a specific attribute attached to it.
+		/// </summary>
+		/// <param name="symbol">The symbol to check for an attribute on</param>
+		/// <param name="compilation">The contextual compilation to lookup types</param>
+		/// <param name="attributeMetadataName">The class of the attribute to check for</param>
+		/// <returns>True if the attribute exists on the symbol, false otherwise</returns>
+		public static bool HasAttribute(
+			this ISymbol symbol,
+			Compilation compilation,
+			string attributeMetadataName
+		) {
+			var attributeSymbol = compilation.GetTypeByMetadataName( attributeMetadataName );

```

Comment:
This is best done once during `CompilationStart`, not within an extension like this for every check. I'd suggest you revert the changes to extracting this extension method to keep the PR focused.